src/osd/PrimaryLogPG.cc

```diff
@@ -3756,31 +3756,38 @@ void PrimaryLogPG::cancel_proxy_write(ProxyWriteOpRef pwop,
 
 class PromoteCallback: public PrimaryLogPG::CopyCallback {
   ObjectContextRef obc;
+  bool may_write;
   PrimaryLogPG *pg;
   utime_t start;
 public:
-  PromoteCallback(ObjectContextRef obc_, PrimaryLogPG *pg_)
+  PromoteCallback(ObjectContextRef obc_, bool may_write_, PrimaryLogPG *pg_)
     : obc(obc_),
+      may_write(may_write_),
       pg(pg_),
       start(ceph_clock_now()) {}
 
   void finish(PrimaryLogPG::CopyCallbackResults results) override {
     PrimaryLogPG::CopyResults *results_data = results.get<1>();
     int r = results.get<0>();
-    pg->finish_promote(r, results_data, obc);
+    pg->finish_promote(r, results_data, may_write, obc);
     pg->osd->logger->tinc(l_osd_tier_promote_lat, ceph_clock_now() - start);
   }
 };
 
 class PromoteManifestCallback: public PrimaryLogPG::CopyCallback {
   ObjectContextRef obc;
+  const bool maybe_write;
   PrimaryLogPG *pg;
   utime_t start;
   PrimaryLogPG::OpContext *ctx;
   PrimaryLogPG::CopyCallbackResults promote_results;
 public:
-  PromoteManifestCallback(ObjectContextRef obc_, PrimaryLogPG *pg_, PrimaryLogPG::OpContext *ctx = NULL)
+  PromoteManifestCallback(ObjectContextRef obc_,
+			  bool maybe_write_,
+			  PrimaryLogPG *pg_,
+			  PrimaryLogPG::OpContext *ctx = NULL)
     : obc(obc_),
+      maybe_write(maybe_write_),
```

Comment:
@tchaikov Current extensible tier (redirect, chunked, dedup) does not support whiteout obj and list snap op. So I think return -ENOTSUPP is the right way if those kind of operations are needed. No need to add this.

Reply:
thanks @myoungwon ! removed and repushed.